src/core/iomgr/iomgr_internal.h

```diff
@@ -46,8 +46,19 @@ typedef struct grpc_iomgr_object {
 int grpc_maybe_call_delayed_callbacks(gpr_mu *drop_mu, int success);
 void grpc_iomgr_add_delayed_callback(grpc_iomgr_closure *iocb, int success);
 
-void grpc_iomgr_register_object(grpc_iomgr_object *obj, const char *name);
```

Comment:
Can we hide the entire mechanism behind an `#ifdef`?

I'm trying to be a little wary of introducing inadvertent string bloat for our debug features.